prow/flagutil/git.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"errors"
+	"flag"
+
+	"k8s.io/test-infra/prow/git/v2"
+	"k8s.io/test-infra/prow/github"
+)
+
+// GitOptions holds options for interacting with git.
+type GitOptions struct {
+	host          string
+	user          string
+	email         string
+	tokenPath     string
+	useSSH        bool
+	useGitHubUser bool
+}
+
+// AddFlags injects Git options into the given FlagSet.
+func (o *GitOptions) AddFlags(fs *flag.FlagSet) {
+	fs.StringVar(&o.host, "git-host", "github.com", "Host to contact for git operations.")
+	fs.StringVar(&o.user, "git-user", "", "User for git commits, optional. Can be derived from GitHub credentials.")
+	fs.StringVar(&o.email, "git-email", "", "Email for git commits, optional. Can be derived from GitHub credentials.")
+	fs.StringVar(&o.tokenPath, "git-token-path", "", "Path to the file containing the git token for HTTPS operations, optional. Can be derived from GitHub credentials.")
+	fs.BoolVar(&o.useSSH, "git-over-ssh", false, "Use SSH when pushing and pulling instead of HTTPS. SSH credentials should be present at ~/.ssh")
```

Comment:
What do we want to do re SSH  host keys, leave the responsibility on the one building the image?